### PR TITLE
fix: Fix to catch error from Promise

### PR DESCRIPTION
**Pull request by MikuroXina**
Fix some bugs about catching an error from `Promise` with `try`-`catch` [reported by sonarcloud](https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=bemusic_bemuse).

### Changelog

Improved error handling code in audio handling code.

### CODE REVIEW

File: bemuse/src/music-previewer/index.js [hunk 1]
```diff
@@ -187,12 +181,8 @@ function createMusicPreviewer() {
...
       play() {
         if (!played) {
-          try {
-            audio.play()
-            played = true
-          } catch (e) {
-            console.warn('Cannot play', audio.src)
-          }
+          audio.play().catch(() => console.warn('Cannot play', audio.src))
+          played = true
```
Comment (dtinth) [inline bemuse/src/music-previewer/index.js:185]: We only want to set `played = true` when `audio.play()` is successful. By putting it outside the try-catch, `played = true` would be set regardless of whether the playing of the audio is successful or not.

```suggestion
          new Promise(r => r(audio.play()))
            .then(() => {
              played = true
            })
            .catch(() => console.warn('Cannot play', audio.src))
```

**Review by dtinth — CHANGES_REQUESTED** [review]
Thank you for the PR! Today I just learned that audio elements’ `.play()` method now return a promise — thanks to your PR. Back in 2016 when this feature is originally implemented, the `.play()` method returns no value, until 2019.

So this change is somewhat breaking, because for people who don't upgrade their OS to a sufficiently new browser, they will get an error like this:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'catch')
```

Also, `try-catch` has been removed. That means if `.play()` happens to throw an error synchronously (in older browsers they do), the music previewer would break.

The code for music previewer has been written quite defensively to try to ensure that it will never throw an error. Hence this change request.

To fix both problems described above, can you replace `goSound.play()` with a Promise wrapper `new Promise(r => r(goSound.play()))`, like this?

```diff
-goSound.play().catch(() => console.warn('Cannot play go sound.'))
+new Promise(r => r(goSound.play())).catch(() => console.warn('Cannot play go sound.'))
```

File: bemuse/src/music-previewer/index.js (hunk 1 above)
Comment (MikuroXina) [reply bemuse/src/music-previewer/index.js]: Fix in 0682fd3c. Sorry for lack of investigation.

File: bemuse/src/music-previewer/index.js [hunk 2]
```diff
@@ -24,6 +24,16 @@ export function preview(url) {
...
 }
 
+function playAudio(element) {
+  return new Promise((resolve, reject) => {
+    try {
+      resolve(element.play())
+    } catch (e) {
+      reject(e)
+    }
+  })
```
Comment (dtinth) [inline bemuse/src/music-previewer/index.js:34]: The Promise constructor already catches errors that is thrown synchronously in the constructor callback, and makes the promise automatically reject. So, it is possible to do this:

```suggestion
  return new Promise((resolve) => {
    resolve(element.play())
  })
```

**Review by dtinth — APPROVED** [review]
Thanks! Nice approach adding a helper function for this. 😃 

It may take a while until I merge this PR since I am not working actively on this project right now, but you can consider your PR accepted. Thanks again!

**Pull request merged**